### PR TITLE
feat(eslint): add more eslint rules

### PR DESCRIPTION
**Pull request by Smrtnyk**
This adds some more eslint rules.
I have tried to add the ones that made sense to me, which can also try to keep code complexity under control.
Please let me know if some these rules are not what you want.

### CODE REVIEW

File: src/plugins/ui-plugin.ts [hunk 1]
```diff
@@ -18,7 +18,10 @@ export default class UIPlugin extends EventEmitter {
 
     render() {}
 
-    setSettings(...args) {}
+    /* eslint-disable-next-line @typescript-eslint/no-unused-vars --
+     unused argument, to keep signature compatible with children
+     */
+    setSettings(settings) {}
```
Comment (pyatyispyatil) [inline src/plugins/ui-plugin.ts:24]: You can rewrite this by function signature, like this
```js
abstract setSettings(...settings: any[]): void
```
And make this class abstract.

**Review by pyatyispyatil — APPROVED** [review]

**Comment by pyatyispyatil** [comment]
Nice rules. The code looks a little fresher now.)

File: src/plugins/timeframe-selector-plugin.ts [hunk 2]
```diff
@@ -300,19 +300,16 @@ export default class TimeframeSelectorPlugin {
         this.dots = dots.sort((a, b) => {
             if (a.pos !== b.pos) {
                 return a.pos - b.pos;
-            } else {
```
Comment (pyatyispyatil) [inline src/plugins/timeframe-selector-plugin.ts:303]: Sometimes I wrote unnecessarily complicated code.(

Comment (Smrtnyk) [reply src/plugins/timeframe-selector-plugin.ts:303]: me too, luckily linters are really good for such things

**Comment by Smrtnyk** [comment]
> Nice rules. The code looks a little fresher now.)

there are few more typescript related ones that would be cool to add, like no-shadow one
But I can make a new pr for that

File: src/plugins/ui-plugin.ts (hunk 1 above)
Comment (Smrtnyk) [reply src/plugins/ui-plugin.ts:24]: > You can rewrite this by function signature, like this
> 
> ```js
> abstract setSettings(...settings: any[]): void
> ```
> 
> And make this class abstract.

tnx, I am not really an OOP guy, for long time working only with functional/procedural programming paradigm
this project is a good refresher on the OOP :D

**Comment by pyatyispyatil** [comment]
Fixes with the UIPlugin class and other things are better done in another PR, yeah.

File: src/plugins/ui-plugin.ts (hunk 1 above)
Comment (pyatyispyatil) [reply src/plugins/ui-plugin.ts:24]: Hmm.
Looks like you can make this method optional in the abstract class.
```ts
abstract class Base {
    abstract required(): void;
    optional?(): string;

    print(): void {
        console.log(this.optional?.() ?? "Base");
    }
}

class Child1 extends Base {
    required(): void { }
}

class Child2 extends Base {
    required(): void { }
    optional(): string {
        return "Child";
    }
}

```

**Pull request merged**